### PR TITLE
OOP hometask done.

### PR DESCRIPTION
**Pull request by kusem**

### CODE REVIEW

File: src/oop/app/src/Transporters/GuzzleAdapter.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+use GuzzleHttp\Client;
+
+class GuzzleAdapter implements TransportInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function getContent(string $url): string
+    {
+        $client = new Client();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Transporters/GuzzleAdapter.php]: please use dependency injection for Client (inject it into constructor)

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+     * @return string
+     */
+    public function getMovieDescription($crawler): string
+    {
+        return trim($crawler->filterXPath('//*[@id="fdesc"]')->text());
+    }
+
+    public function parseContent(string $siteContent)
+    {
+        $this->crawler = new Crawler($siteContent);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: please use dependency injection for Crawler(inject it into constructor)

File: src/oop/app/src/Parsers/FilmixParserStrategy.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+     * @param string $siteContent
+     * @return Movie
+     */
+    public function parseContent(string $siteContent): Movie
+    {
+        $title = $this->getMovieName($siteContent);
+        $poster = $this->getPoster($siteContent);
+        $description = $this->getMovieDescription($siteContent);
+
+        return new Movie($title, $poster, $description);
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/FilmixParserStrategy.php]: please use dependency injection for Movie(inject it into constructor)

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/oop/app/src/Parsers/FilmixParserStrategy.php (hunk 3 above)
Comment (kusem) [reply src/oop/app/src/Parsers/FilmixParserStrategy.php]: > please use dependency injection for Movie(inject it into constructor)

All modifications done.

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php (hunk 2 above)
Comment (kusem) [reply src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: > please use dependency injection for Crawler(inject it into constructor)

All modifications done.

File: src/oop/app/src/Transporters/GuzzleAdapter.php (hunk 1 above)
Comment (kusem) [reply src/oop/app/src/Transporters/GuzzleAdapter.php]: > please use dependency injection for Client (inject it into constructor)

All modifications done.

File: src/oop/app/src/Parsers/FilmixParserStrategy.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+class FilmixParserStrategy implements ParserInterface
+{
+    private Movie $movie;
+
+    /**
+     * Generating Movie class just after class init
+     */
+    public function __construct()
+    {
+        $this->movie = new Movie();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/FilmixParserStrategy.php]: please do not use new keyword, use dependency injection:
public function __construct(Movie $movie)
{
$this->movie= $movie;

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+use src\oop\app\src\Models\Movie;
+use Symfony\Component\DomCrawler\Crawler;
+
+class KinoukrDomCrawlerParserAdapter implements ParserInterface
+{
+    private Crawler $crawler;
+    private Movie $movie;
+
+    public function __construct(){
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: please fix code style for this line

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        $this->crawler = new Crawler();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: please do not use new keyword, use dependency injection:
public function __construct(Crawler $crawler, Movie $movie)
{
$this->crawler = $crawler;
...

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/oop/app/src/Parsers/FilmixParserStrategy.php (hunk 4 above)
Comment (kusem) [reply src/oop/app/src/Parsers/FilmixParserStrategy.php]: > please do not use new keyword, use dependency injection: public function __construct(Movie $movie) { $this->movie= $movie;

In this case, I'd modify ScrapperFactory.php file and pass there a Movie object, am I right?
I am not sure if it's allowed. 
Also, I do not understand what is the point - does it have any sense to throw (empty) objects through different classes?

Comment (yderevianko-cartera) [reply src/oop/app/src/Parsers/FilmixParserStrategy.php]: please read about SOLID and Dependency Injection

File: src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+use src\oop\app\src\Models\Movie;
+use Symfony\Component\DomCrawler\Crawler;
+
+class KinoukrDomCrawlerParserAdapter implements ParserInterface
+{
+    private Crawler $crawler;
+    private Movie $movie;
+
+    public function __construct(Crawler $crawler,Movie $movie){
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Parsers/KinoukrDomCrawlerParserAdapter.php]: code style issue

File: src/oop/app/src/Scrapper.php [hunk 8]
```diff
@@ -7,3 +7,25 @@
...
  */
+
+namespace src\oop\app\src;
+
+class Scrapper
+{
+    private $getBody;
+    private $parseBody;
+
+    public function __construct($getBodyContent, $parseBodyContent)
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Scrapper.php]: please add type hinting

File: src/oop/app/src/Transporters/GuzzleAdapter.php [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace src\oop\app\src\Transporters;
+
+use GuzzleHttp\Client;
+
+class GuzzleAdapter implements TransportInterface
+{
+    private Client $client;
+
+    public function __construct(){
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Transporters/GuzzleAdapter.php]: code style issue

File: src/oop/app/src/Transporters/GuzzleAdapter.php [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        $this->client = new Client();
```
Comment (yderevianko-cartera) [inline src/oop/app/src/Transporters/GuzzleAdapter.php]: please do not use new Client, inject it as dependency

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/oop/app/src/Transporters/GuzzleAdapter.php (hunk 10 above)
Comment (kusem) [reply src/oop/app/src/Transporters/GuzzleAdapter.php]: Done in last commit.

File: src/oop/app/src/ScrapperFactory.php [hunk 11]
```diff
@@ -19,11 +21,11 @@ public function create(string $domain): Scrapper
     {
         switch ($domain) {
             case 'filmix':
-                return new Scrapper(new CurlStrategy(), new FilmixParserStrategy());
+                return new Scrapper(new CurlStrategy(), new FilmixParserStrategy(new Movie()));
             case 'kinoukr':
-                return new Scrapper(new GuzzleAdapter(), new KinoukrDomCrawlerParserAdapter());
+                return new Scrapper(new GuzzleAdapter(), new KinoukrDomCrawlerParserAdapter(new Crawler(), new Movie()));
```
Comment (yderevianko-cartera) [inline src/oop/app/src/ScrapperFactory.php]: You did not pass Client to GuzzleAdapter

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/oop/app/src/ScrapperFactory.php (hunk 11 above)
Comment (kusem) [reply src/oop/app/src/ScrapperFactory.php]: > You did not pass Client to GuzzleAdapter

Sorry, forgot to pass the content of the local file.

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**